core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -182,7 +244,14 @@ internal suspend fun processTransientSettingEvents(
                 }
             }
         }
+        // todo(kc): Enable torch toggle while recording in progress
+        if (currentCameraState.value.videoRecordingState is VideoRecordingState.Active &&
```

Comment:
This will only be applied when other `transientSettings` change. You will want to `combine` the `currentCameraState` with teh `transientSettings` flow so you get events whjen the `videoRecordingState` changes.